test/functional/cursor.test.js

```diff
@@ -3901,6 +3901,50 @@ describe('Cursor', function () {
     });
   });
 
+  describe('Cursor forEach Error propagation', function () {
+    let configuration;
+    let client;
+    let cursor;
+    let collection;
+
+    beforeEach(async function () {
+      configuration = this.configuration;
+      client = configuration.newClient({ w: 1 }, { maxPoolSize: 1 });
+      await client.connect().catch(() => {
+        expect.fail('Failed to connect to client');
+      });
+      collection = client.db(configuration.db).collection('cursor_session_tests2');
+    });
+
+    afterEach(async function () {
+      await cursor.close();
+      await client.close();
+    });
+
+    // NODE-2035
+    it('should propagate error when exceptions are thrown from an awaited forEach call', async function () {
+      const docs = [{ unique_key_2035: 1 }, { unique_key_2035: 2 }, { unique_key_2035: 3 }];
+      await collection.insertMany(docs).catch(() => {
+        expect.fail('Failed to insert documents');
+      });
+      cursor = collection.find({
+        unique_key_2035: {
+          $exists: true
+        }
+      });
+      expect(async function () {
```

Comment:
I don't think you actually need this expect, since if it does throw, the test will fail (mocha tests will fail on any uncaught error, not just "expect"; the expect library is just a handy way to generate errors or sometimes add more information)

you do, however, need to have a .then() before the .catch() where you can put an expect.fail() in case, somehow, the forEach resolves instead of rejecting

Reply:
Thanks for the heads up. That was a good catch.